### ISSUE TITLE
[FEA] Allow for disabling null mapping in Categorify

### ISSUE THREAD

**Issue by kshitizgupta21** [NVIDIA/NVTabular#889]
**Is your feature request related to a problem? Please describe.**
I'm trying to create a customer-merchant interaction DGL graph out of credit card transaction tabular data and using NVTabular for a lot of the data preparation and preprocessing. I'm using NVTabular's `Categorify` to renumber the `card_id` which is the unique identifier for cardholders which will be one kind of nodes in the the graph.  This renumbering of the node ids to continuous indices in the range [0, N) is currently a common and required step in creating a DGL graph. An example is shown below

```
card_id = ['C_ID_123', 'C_ID_234', 'C_ID_456']
# After Categorify ideally I would have card_ids mapped to range [0, N) like this
card_id = [0, 1, 2]
```

**Describe the solution you'd like**
Since currently Categorify always reserves the 0 index for null values, card_id gets mapped to 1,2,..,N instead of 0,1,..N which leads to DGL errors when trying to create the graph. I'd like `Categorify` to give the user flexibility to disable null mapping so that node ids can get mapped to 0,1,..N especially because we are talking about node ids here which we know won't be missing.  If for some reason the user has disabled null mapping but null values show up Categorify should be able to handle those like [OrdinalEnoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OrdinalEncoder.html) does through the `handle_unknown` argument.

**Describe alternatives you've considered**
Currently, using `Categorify` and then `LambdaOp` to subtract 1. Categorify and then subtracting 1 is still kludgy and not simplifying the code in my opinion because this node id renumbering is not a one-off step in the graph creation process. I have to renumber `card_id`, in the cards dataset, `merchant_id` in the merchants dataset, and then both `card_id` and `merchant_id` in the transactions dataset. On top of that, since this what the categories file looks like after  Categorify:

```
	card_id
0	<NA>
1	C_ID_00007093c1
2	C_ID_0001238066
3	C_ID_0001506ef0
4	C_ID_0001793786
```

the inverse transform step i.e. recovering the original `card_id` , `merchant_id` becomes needlessly confusing because now we'd need to add 1 for these id columns but not for the other true categorical columns in the dataset, some of which do have missing values (for which I also use Categorify and it works as intended there).

**Additional context**
Similar functions like Scikit-learn's [Ordinal Encoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OrdinalEncoder.html), cuML's [LabelEncoder](https://docs.rapids.ai/api/cuml/stable/api.html?highlight=label%20encoder#cuml.preprocessing.LabelEncoder.LabelEncoder) do not do implicit null mapping like Categorify and instead give the user the flexibility to handle missing values if they arise through arguments like `handle_unknown`

I think bringing NVTabular's behavior more in line with popular data science libraries like Scikit-learn by allowing the user more flexibility here will ultimately help in adoption of NVTabular for Tabular Data outside of just recommender systems.
